Various fixes to bf-lu

vishwacsena opened the pull request:
Fixes address and enable end to end generated dialogs experience.

feich-ms commented on packages/lu/src/parser/lufile/simpleIntentSection.js:
```diff
@@ -50,13 +50,24 @@ class SimpleIntentSection {
+                    errors.push(BuildDiagnostic({
+                        message: "Invalid utterance definition found. Did you miss a '{' or '}'?",
+                        context: normalIntentStr
+                    }))
```
Only this error will be thrown out? No other error messages?

munozemilio commented on packages/lu/src/parser/lufile/parseFileContents.js:
```diff
@@ -876,37 +879,43 @@ const parseAndHandleSimpleIntentSection = function (parsedContent, luResource) {
-                                        });
-
-                                        throw (new exception(retCode.errorCode.INVALID_INPUT, error.toString(), [error]));
+                                        if (!isChildEntity(entity, entitiesFound)) {
```
This operation is repetitive and can be a function that receives the error message

munozemilio commented:
@vishwac can you please provide what the fixes are about and provide some release notes as well including issues

vishwacsena commented:
@munozemilio sure
- [x] BF-LU parser now throws an error if there are mismatched number of braces in an utterance. 
```
 add {{@AddToQuantity={@number=99}}, {{@AddToQuantity={@number=215}} and {{@AddToQuantity={@number=99}}
```
- [x] Various fixes related to merging as well as validations for new nDepth (hierarchical ML) entity
    - [x] Upon merging (collate), nDepth entities with incompatible types throw
    - [x] Upon merging (collate), nDepth entities are correctly merged via reference
    - [x] Added logic to verify ml entity both at root as well as child level is handled correctly
    - [x] Added validation to ensure every child must have its parent labelled in an utterance
    - [x] Relaxed validations to allow prebuilt child entity type to be labelled in an utterance
    - [x] Relaxed validations to allow list child entity type to be labelled in an utterance
